Objects/bytesobject.c

```diff
@@ -28,6 +28,9 @@ _Py_IDENTIFIER(__bytes__);
 
    Using PyBytesObject_SIZE instead of sizeof(PyBytesObject) saves
    3 bytes per string allocation on a typical system.
+
+   The + 1 accounts for the trailing \0 byte that we include as a safety
+   measure for code that treats the underlying char * as a C string.
 */
 #define PyBytesObject_SIZE (offsetof(PyBytesObject, ob_sval) + 1)
```

Comment:
Thankfully not. Because this code used `offsetof` the macro did not need to change.  The offset never changes.  The existing `+ 1` in the macro is easy to misinterpret as having been there to account for the `[1]` but that isn't how it was used - `PyBytesObject` is always allocated with that one extra byte that is filled with a NUL for C safety for `PyBytes_AsString` users.